common/viperutil/config_util.go

```diff
@@ -24,14 +28,166 @@ import (
 	"github.com/hyperledger/fabric/common/flogging"
 	"github.com/mitchellh/mapstructure"
 	"github.com/pkg/errors"
-	"github.com/spf13/viper"
+	"gopkg.in/yaml.v2"
 )
 
 var logger = flogging.MustGetLogger("viperutil")
 
-type viperGetter func(key string) interface{}
+// prototype for get environment variable
+type envGetter func(key string) interface{}
 
-func getKeysRecursively(base string, getKey viperGetter, nodeKeys map[string]interface{}, oType reflect.Type) map[string]interface{} {
+// ConfigPaths returns the paths from environment and
+// defaults which are CWD and /etc/hyperledger/fabric
+func ConfigPaths() []string {
+	var paths []string
+	if p := os.Getenv("FABRIC_CFG_PATH"); p != "" {
+		paths = append(paths, p)
+	}
+	return append(paths, ".", "/etc/hyperledger/fabric")
+}
+
+// ConfigParser holds the configuration file locations.
+// It keeps the config file directory locations and env. variables
+// From the file the config is unmarshalled and stored.
+// Currently "yaml" is supported.
+type ConfigParser struct {
+
+	// configuration file to process
+	configPaths []string
+	configName  string
+	configFile  string
+
+	// env variable prefix
+	envPrefix string
+
+	// parsed config
+	config map[string]interface{}
+}
+
+// New creates a ConfigParser instance
+func New() *ConfigParser {
+	return &ConfigParser{
+		config: map[string]interface{}{},
+	}
+}
+
+// AddConfigPaths keeps a list of path to search the relevant
+// config file. Multiple paths can be provided.
+func (c *ConfigParser) AddConfigPaths(cfgPaths ...string) {
+	if len(cfgPaths) > 0 {
+		for _, p := range cfgPaths {
+			c.configPaths = append(c.configPaths, p)
+		}
+	}
+}
+
+// SetConfigName assigns the config file name.
+// The extension is not included
+func (c *ConfigParser) SetConfigName(in string) {
+	c.configName = in
+}
+
+// SetConfigFile sets the full config file name.
+// In this case, configPaths search shall not be done
+func (c *ConfigParser) SetConfigFile(in string) {
+	c.configFile = in
+}
+
+// ConfigFileUsed returns the used configFile to fill config store
+func (c *ConfigParser) ConfigFileUsed() string { return c.configFile }
+
+// SetEnvPrefix initializes env variable prefix
+// If "peer" is set here, all environment variable
+// searches shall be prefixed with "PEER_"
+func (c *ConfigParser) SetEnvPrefix(in string) {
+	c.envPrefix = in
+}
+
+// Search for the existence of filename for all supported extensions
+func (c *ConfigParser) searchInPath(in string) (filename string) {
+	var supportedExts []string = []string{"yaml", "yml"}
+	logger.Debugf("Searching for config in ", in)
+	for _, ext := range supportedExts {
+		fullPath := filepath.Join(in, c.configName+"."+ext)
+		logger.Debugf("Checking for ", fullPath)
```

Comment:
Same here. I don't see the need but, if we have it, it should be the correct variant.